src/MyEventEmitter.js

```diff
@@ -1,14 +1,74 @@
+/* eslint-disable max-len */
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+    this.events[eventName].push(listener);
+  }
+
+  once(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.on(eventName, onceListener);
+  }
+
+  off(eventName, listener) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName] = this.events[eventName].filter(
+      (registeredListener) => registeredListener !== listener,
+    );
+  }
+
+  emit(eventName, ...args) {
+    if (!this.events[eventName]) {
+      return;
+    }
+
+    this.events[eventName].forEach((listener) => {
+      listener(...args);
+    });
+  }
+
+  prependListener(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+    this.events[eventName].unshift(listener);
+  }
+
+  prependOnceListener(eventName, listener) {
+    const onceListener = (...args) => {
+      this.off(eventName, onceListener);
+      listener(...args);
+    };
+
+    this.prependListener(eventName, onceListener);
+  }
+
+  removeAllListeners(eventName) {
+    if (eventName) {
```

Comment:
This is a code quality issue. The 'removeAllListeners' method can be simplified. If an event name is provided, you should delete only that event. If no event name is provided, you should delete all events. The current implementation is more complex than it needs to be.